test/Microsoft.AspNet.Razor.Test/Parser/Html/HtmlDocumentTest.cs

```diff
@@ -261,6 +261,20 @@ public void ParseDocumentIgnoresTagsInContentsOfScriptTag()
         }
 
         [Fact]
+        public void ParseDocumentDoesNotRenderExtraNewLine()
+        {
+            ParseDocumentTest("@{\r\n}\r\n<html>",
+                new MarkupBlock(
+                    Factory.EmptyHtml(),
+                    new StatementBlock(
+                        Factory.CodeTransition(),
+                        Factory.MetaCode("{").Accepts(AcceptedCharacters.None),
+                        Factory.Code("\r\n").AsStatement().AutoCompleteWith(null, false),
+                        Factory.MetaCode("}\r\n").Accepts(AcceptedCharacters.None)),
```

Comment:
`\r\n` isn't metacode. It'll be labeled as yellow in the Razor editor if it's left this way. It needs to be split out as a markup span without a `ChunkGenerator` if not nested in a C# block. Otherwise it needs it needs to be a code span.